### PR TITLE
MGMT-8667: CPU and RAM requirements of Master and Worker nodes for Compact and Standard mode respectively added in GetPreflightRequirements

### PR DESCRIPTION
**Pull request by dvachhan**
…PreflightRequirements

# Assisted Pull Request

## List all the issues related to this PR

- [ ] New Feature
- [x] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Assignees

/cc @rewantsoni 
/cc @gobindadas 
/cc @avishayt
/cc @pkliczewski

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] Reviewers have been listed
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- [ ] Are the title and description (in both PR and commit) meaningful and clear?
- [ ] Is there a bug required (and linked) for this change?
- [ ] Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Comment by pkliczewski** [comment]
@jakub-dzon please take a look

File: internal/operators/ocs/ocs_operator.go [hunk 1]
```diff
@@ -185,35 +185,49 @@ func (o *operator) GetHostRequirements(_ context.Context, cluster *common.Cluste
...
-					fmt.Sprintf("%v additional CPUs for each non-boot disk", o.config.OCSPerDiskCPUCount),
-					"At least 3 workers",
-					"At least 1 non-boot disk on 3 workers",
+		}, nil
+	} else {
+		return &models.OperatorHardwareRequirements{
+			OperatorName: o.GetName(),
+			Dependencies: o.GetDependencies(),
+			Requirements: &models.HostTypeHardwareRequirementsWrapper{
+				Worker: &models.HostTypeHardwareRequirements{
```
Comment (jakub-dzon) [inline internal/operators/ocs/ocs_operator.go:217]: What about the master requirements? If the cluster is expected to have masters and workers, the requirements should be specified for both types of nodes. Unless OCS has no additional requirements towards master-type node. Is that the case?

Comment (dvachhan) [reply internal/operators/ocs/ocs_operator.go:217]: In compact mode, the master and worker processes run on the same node so we only have master requirements whereas in standard mode, we only have requirements towards worker nodes.

**Review by jakub-dzon — APPROVED** [review]

**Review by avishayt — COMMENTED** [review]
A couple comments inline.  Also, (1) I guess this should be rebased over the ODF change, and (2) please add ODF to the commit message and PR title

**Pull request closed without merging**